lib/console/gateways/gateway.ex

```diff
@@ -24,8 +25,17 @@ defmodule Console.Gateways.Gateway do
   @doc false
   def changeset(gateway, attrs) do
     gateway
-    |> cast(attrs, [:name, :mac, :public_key, :latitude, :longitude, :team_id, :location])
+    |> cast(attrs, [:name, :mac, :public_key, :latitude, :longitude, :team_id])
     |> validate_required([:name, :mac, :public_key, :latitude, :longitude, :team_id])
+    |> putGeocodingLocation()
     |> unique_constraint(:mac)
   end
+
+  defp putGeocodingLocation(changeset) do
+    case changeset do
+      %Ecto.Changeset{valid?: true, changes: changes} ->
```

Comment:
you can pattern match latitude and longitude out of `changes:`

Reply:
the reason to do that is so that the case won't match if lat/lng aren't present

Reply:
they should get filtered out by the validate_required from line 29 right?